File: database/src/test/java/com/usharik/database/dao/DocumentDatabaseTest.java
package com.usharik.database.dao;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import org.junit.Test;

public class DocumentDatabaseTest {

    @Test
    public void shouldRecreateAfterMigrationFailure_detectsSchemaValidationError() {
        RuntimeException exception = new RuntimeException(
                new IllegalStateException("Migration didn't properly handle: recent_words"));

        assertTrue(DocumentDatabase.shouldRecreateAfterMigrationFailure(exception));
    }

    @Test
    public void shouldRecreateAfterMigrationFailure_detectsMissingMigrationPath() {
        IllegalStateException exception = new IllegalStateException(
                "A migration from 6 to 7 was required but not found.");

        assertTrue(DocumentDatabase.shouldRecreateAfterMigrationFailure(exception));
    }

    @Test
    public void shouldRecreateAfterMigrationFailure_ignoresUnrelatedErrors() {
        RuntimeException exception = new RuntimeException("Disk I/O error");

        assertFalse(DocumentDatabase.shouldRecreateAfterMigrationFailure(exception));
    }
}

File: database/src/test/kotlin/com/usharik/database/dao/DocumentDatabaseTest.kt
package com.usharik.database.dao

import org.junit.Assert.assertEquals
import org.junit.Test

class DocumentDatabaseTest {
    @Test fun databaseNameIsStableForInstalledUsers() {
        assertEquals("quiz-dictionary-database", DocumentDatabase.DB_NAME)
    }

    @Test fun correctionMigrationTargetsVersionEight() {
        assertEquals(7, DocumentDatabase.MIGRATION_7_8.startVersion)
        assertEquals(8, DocumentDatabase.MIGRATION_7_8.endVersion)
    }

    @Test fun scoreMigrationTargetsVersionNine() {
        assertEquals(8, DocumentDatabase.MIGRATION_8_9.startVersion)
        assertEquals(9, DocumentDatabase.MIGRATION_8_9.endVersion)
    }
}
